src/components/Trade/TradeConfirm/TradeConfirm.tsx

```diff
@@ -118,6 +128,43 @@ export const TradeConfirm = () => {
 
   const clearAmounts = useSwapperStore(state => state.clearAmounts)
   const activeSwapper = useSwapperStore(state => state.activeSwapperWithMetadata?.swapper)
+  const updateAffiliateBps = useSwapperStore(state => state.updateAffiliateBps)
+  const updateTradeAmountsFromQuote = useSwapperStore(state => state.updateTradeAmountsFromQuote)
+  const updateFees = useSwapperStore(state => state.updateFees)
+  const defaultAffiliateBps = useSwapperStore(selectSwapperDefaultAffiliateBps)
+
+  const handleDonationToggle = useCallback(async () => {
+    if (!defaultFeeAsset) {
+      throw new Error('No default fee asset')
+    }
+    setIsReloadingTrade(true)
+    const newAffiliateBps = shouldDonate ? '0' : defaultAffiliateBps
+    updateAffiliateBps(newAffiliateBps)
```

Comment:
I see what we're doing now - since the affiliate bps will only ever be the default ones or 0, I guess we could sync `shouldDonate` as a zustand field instead, and have a selector either return the default bps, or 0 if `!shouldDonate`.

This way:

- we would still have the "default affiliate bps" / "affiliate" selectors
- the first one would still return defaults, but the second would get default or else 0 depending on `shouldToggle`

Reply:
Also on the naming of this, `shouldDonate` looks confusingly close to `shouldShowDonationOption`, maybe something something has opted in?

Reply:
Re: naming, inverted logic (as it's opt out, not in) and renamed `shouldDonate` to `hasUserOptedOutOfDonation` in https://github.com/shapeshift/web/pull/4353/commits/5e4ab8cec7decef9409b801a6949860faece12c1.

Reply:
Re: the selector suggestion, see https://github.com/shapeshift/web/pull/4353#discussion_r1181339264 as to why I wish to keep `affiliateBps` a separate, decoupled value.

I agree though, if we were to keep the implementation as if and not seek to support arbitrary bps then we'd want to more tightly couple the `shouldDonate` selection.